utils/vlog/vlog_to_pbtype.py

```diff
@@ -54,68 +54,225 @@
+        list.
+        """
+
+        collected_timings = {}
+
+        # For each path
+        for sdf_cell_path in sdf_cell_paths:
+
+            # Only cell name given
+            if "/" not in sdf_cell_path:
+                sdf_cell_name = sdf_cell_path
+                sdf_inst_name = None
+            # Both cell name and instance path given
+            else:
+                sdf_inst_name, sdf_cell_name = sdf_cell_path.rsplit(
+                    "/", maxsplit=1
+                )
 
-if args.top is not None:
-    top = args.top
-else:
-    wm = re.match(r"([A-Za-z0-9_]+)\.sim\.v", iname)
-    if wm:
-        top = wm.group(1).upper()
-    else:
-        print(
-            "ERROR file name not of format %.sim.v ({}), cannot detect top level. Manually specify the top level module using --top"
-            .format(iname)
-        )
-        sys.exit(1)
+            # Get the SDF timing info for a particular CELL/INSTANCE
+            try:
+                sdf_cell = sdf["cells"][sdf_cell_name]
+            except KeyError:
+                print(
+                    "ERROR, the SDF file does not contain data for cell '{}'".
+                    format(sdf_cell_path)
+                )
+                continue
+
+            # Collect
+            collected_timings.update(sdf_cell[sdf_inst_name])
+
+        return collected_timings
+
+    def pin_list(xml_tag):
+        """
+        Returns a pin list generated from <pb_type> port definition. When the
+        num_pb > 1 ports are named as nameX where X is the pin index.
+        """
+        name = xml_tag.get("name")
+        numb = int(xml_tag.get("num_pins"))
+        if numb == 1:
+            return [name]
+        else:
+            return ["{}{}".format(name, i) for i in range(numb)]
+
+    # Load the SDF
+    with open(sdf_file_name, "r") as fp:
+        sdf = sdfparse.parse(fp.read())
+
+    # Get timescale
+    scale = parse_timescale(sdf["header"]["timescale"])
+
+    # Get pb name
+    pb_name = pb_type_xml.get("name")
+
+    # Get pb ports (inputs, clocks, outputs)
+    pb_inputs = []
+    for xml_tag in pb_type_xml.findall("input"):
+        pb_inputs.extend(pin_list(xml_tag))
+    pb_clocks = []
+    for xml_tag in pb_type_xml.findall("clock"):
+        pb_clocks.extend(pin_list(xml_tag))
+    pb_outputs = []
+    for xml_tag in pb_type_xml.findall("output"):
+        pb_outputs.extend(pin_list(xml_tag))
+
+    # No cell path given, try using the pb_name
+    if sdf_cell_paths is None:
+        sdf_cell_paths = (pb_name.upper())
+
+    # Collect timings for given paths
+    sdf_timings = collect_timings(sdf, sdf_cell_paths)
+    if len(sdf_timings) == 0:
+        exit(-1)
+
+    # Process SDF timing entires
+    for key, sdf_timing in sdf_timings.items():
+
+        sdf_inp = sdf_timing["from_pin"]
+        sdf_out = sdf_timing["to_pin"]
+
+        assert sdf_timing["is_incremental"] == False
+        assert sdf_timing["is_cond"] == False
```

Comment:
It'd be nice if you'd print some message here e.g. "Incremental delays are not supported"

Reply:
Done.